diesel/src/mysql_like/types/mod.rs

```diff
@@ -0,0 +1,105 @@
+//! MySQL specific types
+mod numeric;
+
+use byteorder::WriteBytesExt;
+use std::io::Write;
+
+use deserialize::{self, FromSql};
+use mysql_like::{MysqlLikeBackend};
+use serialize::{self, IsNull, Output, ToSql};
+use sql_types::*;
+
+impl<MysqlLike: MysqlLikeBackend> ToSql<TinyInt, MysqlLike> for i8 {
+    fn to_sql<W: Write>(&self, out: &mut Output<W, MysqlLike>) -> serialize::Result {
+        out.write_i8(*self).map(|_| IsNull::No).map_err(Into::into)
+    }
+}
+
+impl<MysqlLike: MysqlLikeBackend> FromSql<TinyInt, MysqlLike> for i8 {
+    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
+        let bytes = not_none!(bytes);
+        Ok(bytes[0] as i8)
+    }
+}
+
+/// Represents the MySQL unsigned type.
+#[derive(Debug, Clone, Copy, Default, SqlType, QueryId)]
+pub struct Unsigned<ST>(ST);
+
+impl<MysqlLike: MysqlLikeBackend> ToSql<Unsigned<TinyInt>, MysqlLike> for u8 {
+    fn to_sql<W: Write>(&self, out: &mut Output<W, MysqlLike>) -> serialize::Result {
+        ToSql::<TinyInt, MysqlLike>::to_sql(&(*self as i8), out)
+    }
+}
+
+impl<MysqlLike: MysqlLikeBackend> FromSql<Unsigned<TinyInt>, MysqlLike> for u8 {
```

Comment:
There are two ways here to tell the to fix that problem that rustc does not know that the associated type for `RawValue` is `[u8]`

* Explicitly specifying the associated type in the trait bound: `MysqlLike: MysqlLikeBackend<RawValue=[u8]>`
* Us the associated type in place of `[u8]` in the function definition: `fn from_sql(bytes: Option<&MysqlLike::RawValue>) -> deserialize::Result<Self> {`

(I would probably choose the second variant)

Reply:
Thanks for reviewing!

The first snippet there is new to me, I didn't know about associated type bounds, that's very handy.
I think I've tried the second one alone, and it gave me compiler errors about 'not being able to array access ...RawValue ' and maybe the reason it failed to compile was that the compiler couldn't prove that RawValue was [u8] for every MysqlLikeBackend.

So now I'm thinking that I need both of your suggestions, let my give it a try.